backend/de.metas.async/src/main/java/de/metas/async/api/impl/WorkPackageBuilder.java

```diff
@@ -99,22 +104,24 @@ public I_C_Queue_WorkPackage build()
 
 		// Create the workpackage
 		final IWorkPackageQueue workpackageQueue = getWorkpackageQueue();
-		final I_C_Queue_Block queueBlock = getC_Queue_Block();
-		final String queueBlockOriginalTrxName = getTrxName(queueBlock);
+
+		final I_C_Queue_WorkPackage workPackage = initQueueWorkPackage();
+		final String originalWorkPackageTrxName = getTrxName(workPackage);
```

Comment:
u might consider refactoring the second part of this method to `enqueue(workpackage)`